mailbox.c

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright (c) 2012, Broadcom Europe Ltd.
```

Comment:
Where did you get this file from? I am not absolutely sure it is compatible with the MIT License file that we were applying until now to the entire project. We might need to add a note saying that this file has a different license.

Did Broadcom explicitly name a license with this file?